File: src/main/java/com/thoo/api/endpoints/AESEndpoint.java
package com.thoo.api.endpoints;

import com.thoo.api.model.AESKeyFormat;
import com.thoo.api.model.AESModel;
import com.thoo.api.model.BaseModel;
import com.thoo.api.objects.FortniteResponse;
import com.thoo.api.services.AESService;
import com.thoo.api.utils.HttpUtils;
import retrofit2.Call;

public final class AESEndpoint extends Endpoint<AESService> {

    public AESEndpoint(){
        super(AESService.class);
    }

    public FortniteResponse<BaseModel<AESModel>> getAES() {
        Call<BaseModel<AESModel>> call = service.aes();
        return HttpUtils.parse(call);
    }

    public FortniteResponse<BaseModel<AESModel>> getAES(AESKeyFormat format) {
        Call<BaseModel<AESModel>> call = service.aes(format.toString().toLowerCase());
        return HttpUtils.parse(call);
    }

}

File: src/main/java/com/thoo/api/endpoints/AESEndpoint.kt
package com.thoo.api.endpoints

import com.thoo.api.enums.KeyFormat
import com.thoo.api.models.AESModel
import com.thoo.api.models.BaseModel
import com.thoo.api.services.AESService
import com.thoo.api.utils.send
import retrofit2.Retrofit

class AESEndpoint(retrofit: Retrofit) : Endpoint<AESService>(retrofit, AESService::class.java) {

    @JvmOverloads fun getAES(format: KeyFormat = KeyFormat.HEX): BaseModel<AESModel>? = service.aes().send()


}